File: frontend/src/features/tax-profile/types/taxProfile.ts
export interface TaxProfile {
  id: number;
  client_id: number;

  salary_income: boolean;
  business_income: boolean;
  house_property: boolean;
  capital_gain: boolean;
  other_sources: boolean;
  foreign_assets: boolean;

  regime: string;
  financial_year: string;
}

File: backend/app/schemas/tax_profile.py
from pydantic import BaseModel, ConfigDict

class TaxProfileBase(BaseModel):
    salary_income: bool=False
    business_income: bool=False
    house_property: bool=False
    capital_gain: bool=False
    other_sources: bool=False
    foreign_assets: bool=False
    regime:str="NEW"
    financial_year:str="2026-27"

class TaxProfileCreate(TaxProfileBase):
    client_id:int

class TaxProfileUpdate(TaxProfileBase):
    pass

class TaxProfileResponse(TaxProfileBase):
    id:int
    client_id:int

    model_config=ConfigDict(from_attributes=True)
